GetSubscriptionsFeed/__tests__/handler.test.ts

```diff
@@ -0,0 +1,121 @@
+const anEmail = "test@example.com" as EmailString;
+
+const aUserAuthenticationDeveloper: IAzureApiAuthorization = {
+  groups: new Set([UserGroup.ApiMessageRead, UserGroup.ApiMessageWrite]),
+  kind: "IAzureApiAuthorization",
+  subscriptionId: "s123" as NonEmptyString,
+  userId: "u123" as NonEmptyString
+};
+
+const someUserAttributes: IAzureUserAttributes = {
+  email: anEmail,
+  kind: "IAzureUserAttributes",
+  service: {
+    authorizedCIDRs: toAuthorizedCIDRs([]),
+    authorizedRecipients: new Set([]),
+    departmentName: "IT" as NonEmptyString,
+    isVisible: true,
+    maxAllowedPaymentAmount: 0 as MaxAllowedPaymentAmount,
+    organizationFiscalCode: anOrganizationFiscalCode,
+    organizationName: "AgID" as NonEmptyString,
+    requireSecureChannels: false,
+    serviceId: "test" as NonEmptyString,
+    serviceName: "Test" as NonEmptyString,
+    version: 1 as NonNegativeNumber
+  }
+};
+
+const mockAzureApiAuthorization: IAzureApiAuthorization = {
+  groups: new Set(),
+  kind: "IAzureApiAuthorization",
+  subscriptionId: "" as NonEmptyString,
+  userId: "" as NonEmptyString
+};
+
+const mockTableEntry: TableEntry = {
+  RowKey: {
+    _:
+      "1-2-3-4-5-6-e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855"
+  }
+};
+
+const mockQueryEntityResuls: TableService.QueryEntitiesResult<TableEntry> = {
+  entries: [mockTableEntry]
+};
+
+describe("GetSubscriptionFeedHandler", () => {
+  it("should respond error since date now is > available since", async () => {
+    const tableServiceMock = {
+      queryEntities: jest.fn((_, __, ___, f) => {
+        f(undefined, mockQueryEntityResuls, { isSuccessfull: true });
+      })
+    };
+    // use lolex?
+    const todayDate = new Date().toISOString().slice(0, 10);
+
+    const getSubscriptionFeedHandler = GetSubscriptionsFeedHandler(
+      tableServiceMock as any,
+      "subscriptionsFeedTable"
+    );
+
+    const result = await getSubscriptionFeedHandler(
+      aUserAuthenticationDeveloper,
+      undefined as any,
+      someUserAttributes,
+      todayDate
+    );
+
+    expect(tableServiceMock.queryEntities).toHaveBeenCalledTimes(0);
+    expect(result.kind).toBe("IResponseErrorNotFound");
+  });
+
+  it("should respond success if user has subscriptions to services", async () => {
+    const tableServiceMock = {
+      queryEntities: jest.fn((_, __, ___, f) => {
+        f(undefined, mockQueryEntityResuls, { isSuccessful: true });
+      })
+    };
+    // use lolex?
+    const todayDate = "2018-10-20";
+    const getSubscriptionFeedHandler = GetSubscriptionsFeedHandler(
+      tableServiceMock as any,
+      "subscriptionsFeedTable"
+    );
+
+    const result = await getSubscriptionFeedHandler(
+      aUserAuthenticationDeveloper,
+      undefined as any,
+      someUserAttributes,
+      todayDate
+    );
+
+    expect(tableServiceMock.queryEntities).toHaveBeenCalledTimes(3);
+    expect(result.kind).toBe("IResponseSuccessJson");
```

Comment:
We also need to test the result content is what we expect.

Reply:
I don't know the expected behaviour, at the moment I think but I'm not completely sure also an empty array can be a valid result from the API, and in this test is the result content is effectively an empty array. Do you think I can check for that? For more complex result and test also the API should be changed. IMHO.

Reply:
If you have to explain to another person what GetSubscriptionsFeedHandler function MUST do to return the right result probably you say something like:
1) you have to query the db to get all the PROFILE SUBSCRIPTION
2) you have to query the db to get all the SERVICE SUBSCRIPTION
3) you have to query the db to get all the SERVICE UNSUBSCRIPTION
....

What you have to test is that
1) all these steps get correctly executed
2) simulating (mocking) the result of each step the output is what you expect, for example if you simulate that for a user in the X day there is only a SERVICE UNSUBSCRIPTION the return value must include the hashed fiscalcode of the user in the unsubscriptions array.